src/commands/synthetics/reporters/junit.ts

```diff
@@ -0,0 +1,280 @@
+    }
+
+    // Update stats for the suite.
+    suiteRun.$ = {
+      ...suiteRun.$,
+      ...this.getSuiteStats(results, getStats(suiteRun.$)),
+    }
+
+    for (const result of results) {
+      const testSuite: XMLSuite = this.getTestSuite(test, result)
+
+      for (const stepDetail of result.result.stepDetails) {
+        testSuite.testcase.push(...this.getStep(stepDetail))
+      }
+
+      suiteRun.testsuite.push(testSuite)
+    }
+  }
+
+  private getResultStats(result: PollResult, stats: Stats | undefined = getDefaultStats()): Stats {
+    const steps = result.result.stepDetails
+      .map((step) => {
+        if (!step.subTestStepDetails) {
+          return [step]
+        }
+
+        return [step, ...step.subTestStepDetails]
+      })
+      .reduce((acc, val) => acc.concat(val), [])
+
+    for (const step of steps) {
+      const stepStats = this.getStepStats(step)
+
+      stats.tests += stepStats.tests
+      stats.errors += stepStats.errors
+      stats.failures += stepStats.failures
+      stats.skipped += stepStats.skipped
+      stats.allowfailures += stepStats.allowfailures
+      stats.assertions += stepStats.assertions
+      stats.warnings += stepStats.warnings
+    }
+
+    return stats
+  }
+
+  private getStep(stepDetail: Step): XMLStep[] {
+    const mainStep: XMLStep = {
+      $: {
+        allow_failure: stepDetail.allowFailure,
+        is_skipped: stepDetail.skipped,
+        name: stepDetail.description,
+        substep_public_id: stepDetail.subTestPublicId,
+        time: stepDetail.duration / 1000,
+        type: stepDetail.type,
+        url: stepDetail.url,
+        ...this.getStepStats(stepDetail),
+      },
+      browser_error: [],
+      error: [],
+      warning: [],
+    }
+    const steps = [mainStep]
+
+    if (stepDetail.subTestStepDetails) {
+      for (const subStepDetail of stepDetail.subTestStepDetails) {
+        steps.push(...this.getStep(subStepDetail))
+      }
+    }
+
+    if (stepDetail.vitalsMetrics) {
+      mainStep.vitals = stepDetail.vitalsMetrics.map((vital) => ({$: vital}))
+    }
+
+    if (stepDetail.browserErrors?.length) {
+      mainStep.browser_error.push(
+        ...stepDetail.browserErrors.map((error) => ({
+          $: {type: error.type, name: error.name},
+          _: error.description,
+        }))
+      )
+    }
+
+    if (stepDetail.error) {
+      mainStep.error.push({
+        $: {type: 'assertion'},
+        _: stepDetail.error,
+      })
+    }
+
+    if (stepDetail.warnings?.length) {
+      mainStep.warning.push(
+        ...stepDetail.warnings.map((warning) => ({
+          $: {type: warning.type},
+          _: warning.message,
+        }))
+      )
+    }
+
+    return steps
+  }
+
+  private getStepStats(step: Step): Stats {
+    const errors = step.browserErrors ? step.browserErrors.length : 0
+
+    return {
+      allowfailures: step.allowFailure ? 1 : 0,
+      assertions: step.subTestStepDetails ? step.subTestStepDetails.length : 1,
```

Comment:
I am not sure to understand which value `assertions` represents, if it is the number of assertions per step, it should be 0/1 according to the step type and the sum of steps containing an assertion when the step is a subtest.
WDYT?

Reply:
I think we can remove it.
I saw a jUnit that was using it, but I don't think it's that useful for us.